llvm/tools/llvm-readobj/COFFDumper.cpp

```diff
@@ -793,7 +794,10 @@ void COFFDumper::printCOFFDebugDirectory() {
       DictScope PDBScope(W, "PDBInfo");
       W.printHex("PDBSignature", DebugInfo->Signature.CVSignature);
       if (DebugInfo->Signature.CVSignature == OMF::Signature::PDB70) {
-        W.printBinary("PDBGUID", ArrayRef(DebugInfo->PDB70.Signature));
+        SmallString<38> FormattedGUIDStr;
+        raw_svector_ostream OS(FormattedGUIDStr);
```

Comment:
You can stream directly through the ScopedPrinter rather than preparing a string first, try this instead:
  W.startLine() << "PDBGUID: " << fmt_guid(...) << '\n';

Reply:
Thanks for the prompt feedback!

Unfortunately, the suggestion didn't work without first overloading `operator<<` for  `detail:GuidAdapter`. 
``` 
candidate function not viable: no known conversion from 'detail::GuidAdapter' to 'const FormattedBytes'
```

I added the change as a separate commit. Let me know if you prefer I rebase it onto the previous.

Reply:
Yes, that works! I've added the change. 

Let me know if you think we should keep the `operator<<` support or not. I'm inclined to keep it for consistency with the rest of the types.